### PR TITLE
Work on parsing

### PR DESCRIPTION
**Pull request by hnvakil**
Unable to test anything yet due to cmake not building, but first pass at code structure and beginning to fill out functions

### CODE REVIEW

File: src/parser.h [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+ * The zeroth array of strings contains the "rules" section, the first the
+ * "image location" section, the second the "tiles" section.
+ *
+ * @param input_yaml_file The pointer to the file object representing the input
+ * yaml
+ *
+ * @return The array of arrays of strings containing the contents of the file
+ * split into sections
+ */
+char*** get_text_split_sections(FILE* input_yaml_file);
```
Comment (lxbtlr) [inline src/parser.h:27]: Not convinced this needs to be a char***.
A new struct may be appropriate to handle discrete sections of the yaml.

File: src/parser.h [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+ * The values in the return array correspond to:
+ * [0]: 90 degree rotations
+ * [1]: 180 degree rotations
+ *
+ * @param rules_text The block of text containing the rules input, no trailing
+ * empty lines after it.
+ *
+ * @return An array of ints corresponding to rules statuses
+ */
+int* parse_rules(char** rules_text);
```
Comment (lxbtlr) [inline src/parser.h:45]: May be valuable to include the string literals that represent each rule (in yaml there is a discrepancy in the naming scheme)

File: src/parser.h [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+ * images. Return a string corresponding to that location, with a trailing
+ * forward slash.
+ *
+ * @param image_location_text The section of the text containing the image
+ * location text, no trailing lines after it.
+ *
+ * @return The string corresponding to the location of the images with a
+ * trailing forward slash.
+ */
+char* parse_imdir(char** image_location_text);
```
Comment (lxbtlr) [inline src/parser.h:60]: may want to define/specify functionality upon failure (ie no image dir found)

File: src/input.yaml [hunk 4]
```diff
@@ -0,0 +1,18 @@
+---
+#rules
+allow-90-degree-rotations: true
+allow_180-degree-rotations: false
```
Comment (lxbtlr) [inline src/input.yaml:4]: may want to solidify naming scheme

File: src/input.yaml [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+
+#image location
+im_location: "./images"
```
Comment (lxbtlr) [inline src/input.yaml:7]: We need to create an image dir

**Review by lxbtlr — CHANGES_REQUESTED** [review]

File: src/parser.h (hunk 2 above)
Comment (hnvakil) [reply src/parser.h:44]: Agreed, implemented (and put in string literals for the rest of the constant sections in the yaml input)

File: src/input.yaml (hunk 4 above)
Comment (hnvakil) [reply src/input.yaml:4]: Agreed, done

File: src/input.yaml (hunk 5 above)
Comment (hnvakil) [reply src/input.yaml:7]: My thought was the user would provide their own image dir, and here would provide the path to it (we will need to make one for testing)

File: src/parser.h (hunk 3 above)
Comment (hnvakil) [reply src/parser.h]: Agreed, added description of failure behavior to docstring for this function and others. Will implement failure behavior

**Review by zbwrm — APPROVED** [review]

**Pull request merged**